### ISSUE TITLE
How to fuzz a binary project file built by makefile

### ISSUE THREAD

**Issue by songxpu** [eurecom-s3/invscov#2]
Hello andreafioraldi , Invscov is a very interesting idea. I benefit a lot from it. Thank you very much :)
As mentioned in readme, invScov uses the following steps：
1. set the env var INVSCOV_OUTPUT_PATH to an existing empty folder
2. compile the PUT with InvsCov/dump-cc/c++
3. run InvsCov/reconstruct-dump
4. run InvsCov/learn-invariants with the dumper binary produced in the second step
5. run InvsCov/generate-constraints
6. compile with InvsCov/instrument-cc/c++
7. fuzz this last binary with AFLplusplus/afl-fuzz

But I have a small question. If we want to fuzz the target program that is compiled by Makefile,  and there are many project files. In this case, what should I do to fuzz these complex programs? Specifically, I don't know how to do steps 2, 3 and 4 for such a complex project.
I successfully used invscov to fuzz some very simple demos, but I was very confused about using it to fuzz other complex projects.
Looking forward to your reply :)

**Comment by andreafioraldi** [comment]
Hi, I don't fully understand your question.
You have to just take one of the binaries produced by the target build system that you want to fuzz, make cmake or whatever it is.

**Comment by songxpu** [comment]
Hi, Thank you for your reply :)

I am sorry that I did not describe the problem clearly. Now I randomly selected a project to explain my doubts.Like [LightFTP](https://github.com/hfiref0x/LightFTP).

In this program, after executing make, a binary file named fftp can be constructed. After executing the make command, a series of output results will be generated on the command line. It can be seen that the generation of fftp requires multiple executions of gcc or clang commands, and finally multiple files are linked into the final fftp binary file.

The following is part of the output:
```shell
cc  -std=c99 -D_GNU_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -O3 -pedantic -Wall -Wextra -c -fmessage-length=0 -Wno-unused-parameter -Wno-unused-result -fno-ident -v -pthread -MMD -MP -MF"main.d" -MT"main.o" -o "main.o" "../main.c"
...
cc  -std=c99 -D_GNU_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -O3 -pedantic -Wall -Wextra -c -fmessage-length=0 -Wno-unused-parameter -Wno-unused-result -fno-ident -v -pthread -MMD -MP -MF"x_malloc.d" -MT"x_malloc.o" -o "x_malloc.o" "../x_malloc.c"
...
cc  -pthread -o "fftp"  ./cfgparse.o ./fspathtools.o ./ftpconst.o ./ftpserv.o ./main.o ./x_malloc.o   -lpthread -lgnutls

```
Then, I noticed the code implementation of Invscov's dump-cc and instrument-cc files, which essentially executes a clang command.  Therefore, does this mean that if I want to use Invscov and AFL++ to Fuzz the fftp file mentioned above, I must execute files such as dump-cc many times, because normally, only one execution of the clang command cannot successfully build a complete  binary file just like fftp.
